src/rust/ide/view/src/searcher/icons.rs

```diff
@@ -0,0 +1,387 @@
+            let arrow_tip  = arrow_tip.translate_x(6.0.px());
+            let arrow      = arrow_line + arrow_tip;
+
+
+            // === Shape ===
+
+            let shape = rect + arrow;
+            let shape = shape.fill(style.get_color(theme::io));
+            shape.shrink(SHRINK_AMOUNT.px()).into()
+        }
+    }
+}
+
+// A rounded rectangle with the letter "A" and a text cursor.
+mod text_input {
+    use super::*;
+
+    ensogl::define_shape_system! {
+        (style:Style) {
+
+            // === Rectangle ===
+
+            let rect = Rect((16.0.px(),11.0.px())).corners_radius(1.5.px());
+            // Using just the outline.
+            let rect = &rect - rect.shrink(1.0.px());
+            // Creating a gap for the cursor.
+            let gap = Rect((3.0.px(),13.0.px())).translate_x((3.5).px());
+            let rect = rect - gap;
+
+
+            // === Cursor ===
+
+            let cursor_middle = Rect((1.0.px(),15.0.px()));
+            let cursor_top    = Rect((5.0.px(),1.0.px())).translate_y(7.5.px());
+            let cursor_bottom = Rect((5.0.px(),1.0.px())).translate_y((-7.5).px());
+            let cursor        = (cursor_middle + cursor_top + cursor_bottom).translate_x(3.5.px());
+
+
+            // === Letter ===
+
+            // We construct the letter "A", consisting of a diagonal stroke on the left, a diagonal
+            // stroke on the right and a horizontal bar in the middle.
+            let left_stroke   = RoundedLineSegment(((-2.5).px(),(-5.0).px()),1.0.px());
+            let right_stroke  = RoundedLineSegment((2.5.px() ,(-5.0).px()),1.0.px());
+            let bar           = Rect((4.0.px(),1.0.px())).translate_y((-3.5).px());
+            let letter        = left_stroke + right_stroke + bar;
+            let letter        = letter.translate_x((-2.5).px()).translate_y(2.5.px());
+
+
+            // === Shape ===
+
+            let shape = rect + cursor + letter;
+            let shape = shape.fill(style.get_color(theme::io));
+            shape.shrink(SHRINK_AMOUNT.px()).into()
+        }
+    }
+}
+
+// A rounded rectangle with the number "5" and a text cursor.
+mod number_input {
+    use super::*;
+
+    ensogl::define_shape_system! {
+        (style:Style) {
+
+            // === Rectangle ===
+
+            let rect = Rect((16.0.px(),11.0.px())).corners_radius(1.5.px());
+            // Using just the outline.
+            let rect = &rect - rect.shrink(1.0.px());
+            // Creating a gap for the cursor.
+            let gap  = Rect((3.0.px(),13.0.px())).translate_x((3.5).px());
+            let rect = rect - gap;
+
+
+            // === Cursor ===
+
+            let cursor_middle = Rect((1.0.px(),15.0.px()));
+            let cursor_top    = Rect((5.0.px(),1.0.px())).translate_y(7.5.px());
+            let cursor_bottom = Rect((5.0.px(),1.0.px())).translate_y((-7.5).px());
+            let cursor        = (cursor_middle + cursor_top + cursor_bottom).translate_x(3.5.px());
+
```

Comment:
Please refactor such things out. Lets build shapes library here and build icons from them instead.